### PR TITLE
Update Marlin_main.cpp

### PR DESCRIPTION
**Pull request by barus93**
Y home issue with dual x carriage
if def DUAL_X_CARRIAGE home only with x direction

### CODE REVIEW

File: Marlin/Marlin_main.cpp [hunk 1]
```diff
@@ -1502,8 +1502,8 @@ static void homeaxis(AxisEnum axis) {
 
     int axis_home_dir;
 
-    #ifdef DUAL_X_CARRIAGE
-      if (axis == X_AXIS) axis_home_dir = x_home_dir(active_extruder);
+    #if defined(DUAL_X_CARRIAGE) && (axis == X_AXIS)
```
Comment (nophead) [inline Marlin/Marlin_main.cpp:1505]: You can't test the run time variable axis with a preprocessor #if

Comment (nophead) [inline Marlin/Marlin_main.cpp:1505]:
```
#ifdef DUAL_X_CARRIAGE
     axis_home_dir = axis == X_AXIS ?  x_home_dir(active_extruder)  :  home_dir(axis);
#else
```

**Comment by barus93** [comment]
sorry, in my printer i run this code

```
#ifdef DUAL_X_CARRIAGE
      if (axis == X_AXIS) 
      axis_home_dir = x_home_dir(active_extruder);
      else
      axis_home_dir = home_dir(axis);
      #else
      axis_home_dir = home_dir(axis);
    #endif
```

but I'm not a good arduino programmer..

**Comment by nophead** [comment]
That would work, why did you push something else?

**Comment by barus93** [comment]
I was looking for a better solution and i thought its work..

**Comment by nophead** [comment]
You should only push code you have actually tested in your printer not test one thing and push another.

**Comment by tonnico** [comment]
You could make something like this.

```
#ifdef DUAL_X_CARRIAGE
    if (axis == X_AXIS) 
        axis_home_dir = x_home_dir(active_extruder);
    else
#endif
axis_home_dir = home_dir(axis);
```

**Comment by thinkyhead** [comment]
#1958 is a redo of this, and should fix the issue!

**Pull request closed without merging**